src/charmed_kubeflow_chisme/pebble/_update_layer.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import traceback
+from logging import Logger
+
+from ops.model import BlockedStatus, Container
+from ops.pebble import ChangeError, Layer
+
+from ..exceptions import ErrorWithStatus
+
+
+def update_layer(container_name: str, container: Container, new_layer: Layer, logger: Logger):
+    """Updates the Pebble configuration layer if changed."""
```

Comment:
I'd add some docstrings with an `Args` section and what this method raises on failure.